gcovr/decision_analysis.py

```diff
@@ -208,7 +209,7 @@ def continue_multiline_decision_analysis(self, lineno: int, code: str) -> None:
             # false is the delta between executed blocks and executions of the decision statement.
             last_decision_line_cov.decision = DecisionCoverageConditional(
                 exec_count,
-                last_decision_line_cov.count - exec_count,
+                max(last_decision_line_cov.count - exec_count, 0),
```

Comment:
Yes this fixes the failing assertion, but if this count is negative that also means that our expectations about the control flow are wrong. In that case, do you think it's better to create a DecisionCoverageConditional(exec_count, 0) or should a DecisionCoverageUncheckable be generated?

Reply:
You are absolutely right. I thought about this but hadn't enough time to go into detail to find the cause for this misconception. It fixed the problem for now, and the generated decision results for my application seem reasonable. But I can go into more detail if you think this feature could still stand improvement.

Reply:
I prefer also a `DecisionCoverageUncheckable` instead of a count of 0.